### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by jrosario19**
### IMPORTANT: Please refer to our comment on the last code review

- Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (FrederickMih) [inline app/models/user.rb:5]: - [ ] Hi team, the console error on the screenshot below hinders me from setting up your project completely. Hence I couldn't register or login into your app. If the error is from my end, I'll be happy to connect with you guys to resolve else make sure to fix it and your app should be working as intended. 


Screenshot from 2022-08-24 19-37-48



Thanks :pray:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- PR is professional :+1: 
- Descriptive READM.md file :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (jrosario19) [reply app/models/user.rb:5]: Hello @FrederickMih 

Thank you for your review and recommendations. We tested the project in both platforms (Windows and Ubuntu) and the problem is related with the bootsnap gem as follows:
- In order for us to be able to run the project successfully in windows platform we have to change the bootsnap gem to `gem 'bootsnap', '>= 1.1.0', '< 1.4.2', require: false`.
- In order for us to be able to run the project successfully in Ubuntu platform we have to change the bootsnap gem to `gem "bootsnap", "~> 1.13", require: false`.

As you are using Ubuntu try installing the last gem, with the following commands:
```
bundle remove bootsnap
bundle add bootsnap
```
IMPORTANT: After installing the gem you should write `require:false` at the end of the line of bootsnap gem.

**Review by addod19 — APPROVED** [review]
### Hi @jrosario19, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job fixing the bug about `bootsnap`, it is really professional adding how to fix it in the README :100: 
Screenshot from 2022-08-24 21-42-58



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (FrederickMih) [reply app/models/user.rb:5]: Hi @jrosario19 .


Thanks 🙏 so much

**Pull request merged**